### DISCUSSION TITLE
field!: int not support

### DISCUSSION THREAD

**Discussion by sword-jin** [cue-lang/cue#1724] [Q&A] [answered]
I found this `discusstion` https://github.com/cue-lang/cue/discussions/441

```
ntroduce ! as a field marker, and introduce the following rules, w.r.t. exporting:

field: int: export only if concrete
field!: int: optional, but needs to be explicitly declared for a complete schema to be valid.
field?: 1: always optional, don't export, even if concrete.
```

**Comment by mvdan** [comment, marked as answer]
This feature is called required fields, and it is proposed in https://github.com/cue-lang/cue/issues/822. Note that it hasn't been approved nor implemented yet. I'm marking this question as resolved, because that's the reason why it doesn't work yet :)

**Reply by sword-jin** [reply]
sure
